google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/TraceUtil.java

```diff
@@ -69,10 +97,44 @@ static void endSpanWithFailure(Span span, SpannerException e) {
     span.end();
   }
 
+  /**
+   * Appends a list of span names.
+   *
+   * <p>If called multiple times the library keeps the list of unique span names from all the calls.
+   *
+   * @param spans list of span names.
+   */
   static void exportSpans(String... spans) {
-    SampledSpanStore store = Tracing.getExportComponent().getSampledSpanStore();
-    if (store != null) {
-      store.registerSpanNamesForCollection(Arrays.asList(spans));
+    Preconditions.checkNotNull(spans);
+    for (String spanName : spans) {
+      RecordEventsSpanImpl recordEventsSpan = createSampledSpan(spanName);
+      recordEventsSpan.end(EndSpanOptions.builder().setSampleToLocalSpanStore(true).build());
     }
   }
+
+  /**
+   * Create span to records trace events.
+   *
+   * @param spanName span names.
+   */
+  private static RecordEventsSpanImpl createSampledSpan(String spanName) {
+    Random random = new Random();
+    SpanId parentSpanId = SpanId.generateRandomId(random);
+    SpanContext sampledSpanContext =
+        SpanContext.create(
+            TraceId.generateRandomId(random),
+            SpanId.generateRandomId(random),
+            TraceOptions.builder().setIsSampled(true).build(),
+            Tracestate.builder().build());
+    return RecordEventsSpanImpl.startSpan(
+        sampledSpanContext,
+        spanName,
+        Span.Kind.CLIENT,
+        parentSpanId,
+        false,
+        TraceParams.DEFAULT,
+        recordEventSpan,
+        null,
+        TestClock.create());
+  }
```

Comment:
@suraj-qlogic Can you please run this with **cloud spanner** and check how many extra spans it records. ?

Because while initialization all the spans will be registered and there will be tracing for each operation irrespective of it is actually being called.